sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java

```diff
@@ -215,6 +224,83 @@ private void createRunnerAndConsumersForPTransformRecursively(
    */
   public BeamFnApi.InstructionResponse.Builder processBundle(BeamFnApi.InstructionRequest request)
       throws Exception {
+    BeamFnApi.ProcessBundleResponse.Builder response = BeamFnApi.ProcessBundleResponse.newBuilder();
+
+    BundleProcessor bundleProcessor =
+        bundleProcessors.computeIfAbsent(
+            request.getProcessBundle().getProcessBundleDescriptorId(),
```

Comment:
I like where this is going but we can't run a DoFn instance in parallel (since this is what we promise to users) so this lookup map needs to contain a list of BundleProcessor instances that are free. If there isn't one that is free then we need to create one.

Also if a BundleProcessor fails during execution we need to throw it away since the DoFn's may be in a bad state.